### PR TITLE
Optimize ambiguous parser rules

### PR DESCRIPTION
**Pull request by temanbrcom**

### CODE REVIEW

**Review by grianbrcom — APPROVED** [review]
:flamingo:

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CICSParser.g4 [hunk 1]
```diff
@@ -1046,7 +1046,7 @@ idms_only_words
     | WCC | WHITE | WITHIN | YELLOW
     ;
 
-cicsWord: NONNUMERICLITERAL | NUMERICLITERAL | integerLiteral | generalIdentifier | cobolWord | cics_cobol_intersected_words;
+cicsWord: NONNUMERICLITERAL | NUMERICLITERAL | integerLiteral | generalIdentifier | cics_cobol_intersected_words;
```
Comment (asatklichov) [inline server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CICSParser.g4:1049]: Not sure if removal of cobolWord will cause a fail, at least on some tests?

**Review by asatklichov — APPROVED** [review]
🦺

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CICSParser.g4 (hunk 1 above)
Comment (temanbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CICSParser.g4:1049]: `cobolWord` is indirectly covered by the `generalIdentifier`

**Pull request merged**